static/js/external/Makefile

```diff
@@ -0,0 +1,61 @@
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Variables ###################################################################
+
+GET = wget -qO
+CDNJS = https://cdnjs.cloudflare.com/ajax/libs
+
+
+# Files #######################################################################
+
+all: angular.min.js angular.min.js.map angular-mocks.js angular-route.min.js angular-route.min.js.map \
+	 angular-ui-router.min.js jquery.min.js jshint.js restangular.min.js underscore-min.js \
+	 underscore-min.map
+
+angular.min.js:
+	$(GET) $@ $(CDNJS)/angular.js/1.4.6/angular.min.js
```

Comment:
These rules have the disadvantage that they won't reload the files when the version was changed in the Makefile.  So if upstream bumps the version, you pull and run `make <whatever>`, you will still have the old versions of the JS files, since make can't detect they are not up to date.

A proper solution to this problem would be to cache the version (or URL) used for each file, and redownload it if the URL changed, but this would require a bit of coding.  A simpler solution that doesn't get all cases right would be to use

    GET = wget -qN

above, remove `$@` in all rules, create a `.FORCE` target

    .PHONY: .FORCE

and add it as a dependency to all rules.  Make will run the rules every time now, and wget will compare the timestamp of the local files with the remote file, and redownload it if the remote file is newer (so downgrading will likely fail, while upgrading to newer versions should work correctly).

(You could also use `wget -qNO` and leave `$@` in.  However, `-N` disables the numbered backups anyway, so the output filename will be set correctly automatically.)